packages/golden_toolkit/test/multi_screen_golden_test.dart

```diff
@@ -131,6 +131,78 @@ Future<void> main() async {
         expect(tester.binding.window.textScaleFactor, equals(initialTextScaleFactor));
         expect(tester.binding.window.padding, equals(initialViewInsets));
       });
+
+      testGoldens('Should expand scrollable if autoHeight is true', (tester) async {
+        await tester.pumpWidgetBuilder(ListView.builder(
+          itemBuilder: (BuildContext context, int index) {
+            return Container(
+              color: Colors.black.withRed(255 - index * 5),
+              height: 50,
+            );
+          },
+          itemCount: 40,
+        ));
+
+        await multiScreenGolden(
+          tester,
+          'auto_height_list_view',
+          autoHeight: true,
+          devices: [
+            const Device(
+              name: 'anything',
+              size: Size(100, 200),
+              brightness: Brightness.light,
+            )
+          ],
+          skip: !Platform.isMacOS,
+        );
+      });
+
+      testGoldens('Should expand scrollable only if not infinite', (tester) async {
+        await tester.pumpWidgetBuilder(ListView.builder(
+          itemBuilder: (BuildContext context, int index) {
+            return Container(
+              color: Colors.black.withRed(255 - index * 5),
+              height: 50,
+            );
+          },
+        ));
+
+        await multiScreenGolden(
+          tester,
+          'auto_height_infinite_list_view',
+          autoHeight: true,
+          devices: [
+            const Device(
+              name: 'anything',
+              size: Size(100, 200),
+              brightness: Brightness.light,
+            )
+          ],
+          skip: !Platform.isMacOS,
+        );
+      });
+
+      testGoldens('Should shrink to finders height if autoHeight is true', (tester) async {
```

Comment:
not sure I have the right answer here, but I wonder if shrinking the content is the expected behavior. 

i.e. imagine I was trying to simulate what a screen looked like on a a particular device size, but the content was shorter than the screen. 

With this solution, it would shrink to be smaller than the device dimensions.

I wonder if the semantics of this should be to fit to the screen size and expand if necessary?

@mZadorskii @aanelson do you guys have any thoughts here?

@christian-muertz what do you think? Do you have a use case for auto shrinking to smaller than the supplied device size?

Reply:
> Do you have a use case for auto shrinking to smaller than the supplied device size?

Sometimes we just want to make sure a widget looks correct at the device's width. Not shrinking it would add a lot of white to the bottom.

Imagine you just want to check the appearance of an app bar which height you don't already know. This allows shrinking the golden file to the final height.

Reply:
That use case makes sense to me... I think the piece I'm struggling with is the semantic difference of a golden that validates a particular widget vs. a "screen". I think a screen should remain anchored in the notion of a device.

In terms of my original question/concern.. I think we can ignore it for now. If someone wants to capture a screen with content shorter than the configured device's height, then they can just leave autoHeight: false so they can visualize the white space.

This and your other comment make me think that longer term, we should ultimately tweak the semantics of the APIs to more clearly cater to the following use cases:

- capture one or more goldens in the context of a device screen (e.g. multiScreenGolden)
- capture a golden of a widget independent of a device, and allow auto sizing/shrinking along both axes. Conceptually, this would simply be a wrapper over flutter test's built-in golden assertion that adds the auto sizing capability

None of that needs to be addressed now. I'm just thinking out loud... I have an outstanding mental to-do to move these assertion APIs to be extension methods on WidgetTester, and that will be a time to fix some of the problems with the current APIs.

Reply:
Maybe this could even go as far as creating something like `GoldenSizeConstraint`s:

```dart
final GoldenSizeConstraint iphoneXsList = GoldenSizeConstraint(
  minWidth: 300,
  maxWidth: 300,
  minHeight: 500,
  maxHeight: 2000,
);
```

This would allow for example an infinite scrolling list to expand to a limited amount.

Reply:
yeah, I think a constraints based solution is the right long term answer. For now, I think we should merge your PR in. 

I think we should incorporate this more robust model into the next version of this API.